README.md

```diff
@@ -1,278 +1,313 @@
-| `key_column`  | The table column specified as key_column when the aggregating index was created. See [Firebolt docs](https://docs.firebolt.io/sql-reference/commands/ddl-commands#create-join-index) for more information |
-| `aggregation` | One or more aggregation functions performed as part of the index                                                                                                                          |
+### Model Configuration for Fact Tables
 
-#### Example of fact table with aggregating index
-```sql
--- orders.sql
-{{
-  config(
-    materialized = 'table',
-    table_type = 'fact',
-    primary_index = 'id',
+A dbt model can be created as a Firebolt fact table and configured using the following syntax:
+
+Project file syntax
+```yml
+models:
+  <resource-path>:
+    +materialized: table
+    +table_type: fact
+    +primary_index: [ <column-name>, ... ]
+    +indexes:
+      - type: aggregating
+        key_column: [ <column-name>, ... ]
+        aggregation: [ <agg-sql>, ... ]
+      ...
+```
+
+Property file syntax
+```yml
+models:
+  - name: <model-name>
+    config:
+      materialized: table
+      table_type: fact
+      primary_index: [ <column-name>, ... ]
+      indexes:
+        - type: aggregating | join
+          key_column: [ <column-name>, ... ]
+          aggregation: [ <agg-sql>, ... ]
+        ...
+```
+
+Config block syntax
+```
+{{ config(
+    materialized = "table"
+    table_type = "fact"
+    primary_index = [ "<column-name>", ... ],
     indexes = [
       {
-        'type': 'aggregating',
-        'key_column': 'order_id',
-        'aggregation': ['COUNT(DISTINCT status)', 'AVG(customer_id)']
-      }
+        type = "aggregating"
+        key_column = [ "<column-name>", ... ],
+        aggregation = [ "<agg-sql>", ... ],
+      },
+      ...
     ]
-    )
-}}
+) }}
 ```
-#### Join index
 
-|       key        |                                       value                                        |
-|------------------|------------------------------------------------------------------------------------|
-| `type`             | `join`                                                                               |
-| `join_column`      | The join_column specified when the join index was created.                         |
-| `dimension_column` | One or more dimension columns that were specified when the join index was created. |
+#### Description of Configurations Pertaining to Fact Tables
 
+| Configuration     | Description                                                                               |
+|-------------------|-------------------------------------------------------------------------------------------|
+| `materialized`    | How the model will be materialized into Firebolt. Must be `table` to create a fact table. |
```

Comment:
Isn't it the case that it must be `table` period? I thought for now we weren't supporting `view`?

Reply:
AFAIK we do support ephemeral materializations as well. They are manifested as CTEs in Firebolt. @swanderz fact check please